Skip shader compile status check on ANGLE

lexaknyazev opened the pull request:
#### bcacfd6ffbd07a582adc2af9a4ceae429e14eb1c
<pre>
Skip shader compile status check on ANGLE
<a href="https://bugs.webkit.org/show_bug.cgi?id=241010">https://bugs.webkit.org/show_bug.cgi?id=241010</a>

Reviewed by Kimmo Kinnunen.

* To not stall the pipeline, do not query shader compile
status immediately after compilation when ANGLE is used.

* Removed the test that expects shader compilation logs
to be printed to the console.

Drive-by:
* Return true from InspectorShaderProgram::updateShader when
compiling or linking fail to avoid Inspector UI crash on
live-editing shaders in debug builds. Update the test.

* LayoutTests/fast/canvas/webgl/shader-compile-logging-expected.txt: Removed.
* LayoutTests/fast/canvas/webgl/shader-compile-logging.html: Removed.
* LayoutTests/inspector/canvas/updateShader-webgl-expected.txt:
* LayoutTests/inspector/canvas/updateShader-webgl.html:
* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp:
(WebCore::WebGLRenderingContextBase::compileShader):
* Source/WebCore/html/canvas/WebGLShader.cpp:
(WebCore::WebGLShader::WebGLShader):
* Source/WebCore/html/canvas/WebGLShader.h:
* Source/WebCore/inspector/InspectorShaderProgram.cpp:
(WebCore::InspectorShaderProgram::updateShader):

Canonical link: <a href="https://commits.webkit.org/253593@main">https://commits.webkit.org/253593@main</a>
</pre>
























<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/de9b4d241e2894e5c1b0e00f3a2a4f3950930dd7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/86475 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/30403 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/17441 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/95323 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/149036 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/90463 "Passed tests") | [✅ ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/64/builds/28823 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/25374 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/78635 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/90568 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/92091 "Passed tests") | [✅ ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/64/builds/28823 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/73439 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/23420 "Passed tests") | 
| | [✅ ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/64/builds/28823 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/78757 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/66429 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/66/builds/26708 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/12607 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/67/builds/26622 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/13620 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2561 "Built successfully and passed tests") | [✅ ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/63/builds/28301 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/73439 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/65/builds/28242 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/32902 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

kkinnunen-apple commented on Source/WebCore/inspector/InspectorShaderProgram.cpp:
```diff
@@ -123,10 +123,9 @@ bool InspectorShaderProgram::updateShader(Inspector::Protocol::Canvas::ShaderTyp
                         contextWebGLBase.shaderSource(*shader, source);
                         contextWebGLBase.compileShader(*shader);
-                        if (shader->isValid()) {
+                        if (std::get<bool>(contextWebGLBase.getShaderParameter(*shader, GraphicsContextGL::COMPILE_STATUS)))
```
Technically, this might return non-bool, for example when the context times out just for this call.
So maybe first ask if the variant is bool?

kkinnunen-apple commented on Source/WebCore/html/canvas/WebGLShader.h:
```diff
@@ -52,7 +54,9 @@ class WebGLShader final : public WebGLSharedObject {
     GCGLenum m_type;
     String m_source;
+#if !USE(ANGLE)
     bool m_isValid;
```
You might initialise this to false, so then you avoid one ifdef chunk in the constructor

lexaknyazev committed:
Skip shader compile status check on ANGLE
https://bugs.webkit.org/show_bug.cgi?id=241010

Reviewed by Kimmo Kinnunen.

* To not stall the pipeline, do not query shader compile
status immediately after compilation when ANGLE is used.

* Removed the test that expects shader compilation logs
to be printed to the console.

Drive-by:
* Return true from InspectorShaderProgram::updateShader when
compiling or linking fail to avoid Inspector UI crash on
live-editing shaders in debug builds. Update the test.

* LayoutTests/fast/canvas/webgl/shader-compile-logging-expected.txt: Removed.
* LayoutTests/fast/canvas/webgl/shader-compile-logging.html: Removed.
* LayoutTests/inspector/canvas/updateShader-webgl-expected.txt:
* LayoutTests/inspector/canvas/updateShader-webgl.html:
* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp:
(WebCore::WebGLRenderingContextBase::compileShader):
* Source/WebCore/html/canvas/WebGLShader.cpp:
(WebCore::WebGLShader::WebGLShader):
* Source/WebCore/html/canvas/WebGLShader.h:
* Source/WebCore/inspector/InspectorShaderProgram.cpp:
(WebCore::InspectorShaderProgram::updateShader):

Canonical link: https://commits.webkit.org/253593@main
